src/sql/src/plan.rs

```diff
@@ -1298,16 +1298,52 @@ pub struct Ingestion {
 pub struct WebhookValidation {
     /// The expression used to validate a request.
     pub expression: MirScalarExpr,
+    /// Description of the source that will be created.
+    pub relation_desc: RelationDesc,
     /// The column index to provide the request body and whether to provide it as bytes.
     pub bodies: Vec<(usize, bool)>,
     /// The column index to provide the request headers and whether to provide the values as bytes.
-    ///
-    /// TODO(parkmycar): Support filtering down to specific headers.
     pub headers: Vec<(usize, bool)>,
     /// Any secrets that are used in that validation.
     pub secrets: Vec<WebhookValidationSecret>,
 }
 
+impl WebhookValidation {
+    const MAX_REDUCE_TIME: Duration = Duration::from_secs(2);
+
+    /// Attempt to reduce the internal [`MirScalarExpr`] into a simpler expression.
+    ///
+    /// The reduction happens on a separate thread, we also only wait for
+    /// [`WebhookValidation::MAX_REDUCE_TIME`] before timing out and returning an error.
+    pub async fn reduce_expression(&mut self) -> Result<(), &'static str> {
+        let WebhookValidation {
+            expression,
+            relation_desc,
+            ..
+        } = self;
+
+        // On a different thread, attempt to reduce the expression.
+        let mut expression_ = expression.clone();
+        let desc_ = relation_desc.clone();
+        let reduce_task = mz_ore::task::spawn_blocking(
```

Comment:
note that this will not be shutdown once the tokio timeout happens, it will finish and do nothing with the result. this means we could eat into cpu here

Reply:
Yeah, and AFAIK there isn't a way to shutdown the thread, right?

My main motivation here was to not block the coordinator, so I personally would be okay with this tradeoff, but open to other thoughts!

Reply:
posix pretends there is a way, but its all smoke and mirrors!

avoiding blocking the coordinator with this seems fine!